config/env/types.go

```diff
@@ -0,0 +1,213 @@
+package env
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// matches a variable surrounded by curly braces with leading dollar sign.
+// ex: ${MY_VAR}
+var regex = regexp.MustCompile(`\${[A-Z]+[A-Z1-9_]*}`)
+
+// EnvironmentError corresponds with a missing environment variable
+type EnvironmentError struct {
+	varName string
+}
```

Comment:
Not, something I'm asking to change; there is no memory impact. It's just more stylistic.  I would have just used the type here, and not the struct. 
So, for the EnvironmentError:
```
type EnvironmentError string

func (ee EnvironmentError) Error() string {
         return "environment variable "+string(ee)+" not found"
}
```

I only use a struct if there is going to be more items for the error.